### PR TITLE
Update ug

### PR DESCRIPTION
**Pull request by tanjieling0**
* Refactor data contraints
* Reformat planned enhancements

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -257,10 +260,8 @@ Find by role example:
 
 Find by remark example:
 * `find r/` returns all persons who have an empty remark.
-* `find r/has a dog` returns all persons who have the remark `has a dog`.
-* `find r/dog` returns all persons who have the remark `has a dog`.
-* `find r/a has` returns all persons who have the remark `has a dog`.
-
+* `find r/has a dog` returns all persons who have the remark containing words `has a dog`.
```
Comment (purivirakarin) [inline docs/UserGuide.md]: image

According to this, shall we change this `find r/has a dog returns all persons who have the remark containing words has a dog.` into `find r/has a dog returns all persons who have the remark containing all these words ['has', 'a', 'dog']`

File: docs/UserGuide.md [hunk 2]
```diff
@@ -98,6 +99,26 @@ e.g. in `add n/NAME`, `NAME` is a parameter which can be used as `add n/John Doe
 
 <section id="help">
 
+# Data Constraints
```
Comment (shaunlxw) [inline docs/UserGuide.md]:
```suggestion
## Data Constraints
```

**Review by purivirakarin — CHANGES_REQUESTED** [review]
Thanks for the changes, there is a question regarding the `find` command.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -113,20 +134,24 @@ Format: `help`
 
 ## Adding a person: `add`
 
-Adds a person (Client, Supplier or Employee) to the address book. Note that each role (eg. Client, Supplier, Employee) has its own specific set of fields that can be added.
+Adds a person (Client, Supplier or Employee) to the address book. Note that each role (e.g. Client, Supplier, Employee) has its own specific set of fields that can be added.
+* The input for all fields should adhere to the [Data Constraints](#data-constraints).
```
Comment (shaunlxw) [inline docs/UserGuide.md]: Maybe it doesn't need to be a bullet point and can be in the same paragraph?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -257,10 +260,8 @@ Find by role example:
 
 Find by remark example:
 * `find r/` returns all persons who have an empty remark.
-* `find r/has a dog` returns all persons who have the remark `has a dog`.
-* `find r/dog` returns all persons who have the remark `has a dog`.
-* `find r/a has` returns all persons who have the remark `has a dog`.
-
+* `find r/has a dog` returns all persons who have the remark containing words `has a dog`.
+*
```
Comment (shaunlxw) [inline docs/UserGuide.md]: Should be empty line?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -348,33 +349,38 @@ Retrieve information on a group of profiles at once with this function! This can
...
-* After entering `export`, you can choose the filename. The filename must end with `.csv`. If you do not specify a filename, it will be automatically named as `contact.csv`
+* Following `export` in Step 2, you can choose the filename to save the file as. The filename must end with `.csv`. If you do not specify a filename, it will be automatically named as `contact.csv`
+
+Example:
 
-Example
 Step 1: `list` first returns the list of all contacts
-Step 2: `export` a CSV file containing all contacts to a folder on your laptop that is located within the same directory as the NetConnect application. 
-  
-![result for 'export all'](images/exportall.png)
```
Comment (shaunlxw) [inline docs/UserGuide.md:362]: Why are we removing the screenshot?

File: docs/UserGuide.md [hunk 6]
```diff
@@ -408,13 +414,17 @@ If your changes to the data file makes its format invalid, NetConnect will disca
...
+# Planned Enhancements
 The NetConnect team is working on new features and fixes for you, but they are unfortunately unavailable in this current implementation. We intend to have future fixes for these occurences below!
 
-1. When invoking showrelated on a id that does not exist, the error message will be `Contact not found` instead of `0 persons listed`.
+1. Improved error messages for `showrelated` command
+2. Improved command success messages for `relate` and `unrelate` commands
+3. Improved validity checks for `relate` and `unrelate` commands
+4. Truncate text in GUI to accommodate long text fields
+5. Handle situations where input entered is the same as the current value for `edit` command
+6. Data recovery
```
Comment (shaunlxw) [inline docs/UserGuide.md]: Minor but [Markdown conventions](https://se-education.org/guides/conventions/markdown.html) suggests using
```
1.
1.
```

for ordered list

**Review by shaunlxw — CHANGES_REQUESTED** [review]
LGTM less some minor formatting changes.

File: docs/UserGuide.md [hunk 7]
```diff
@@ -206,19 +211,17 @@ Format: `edit i/ID [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [r/remark] [t/TAG]
 * Edits the person with the specified `ID`. `ID` refers to the unique identification number assigned to each person when first added to the list.
 * `ID` **must refer to a person that exist within NetConnect**.
 * At least one of the optional fields `[..]` must be provided.
```
Comment (purivirakarin) [inline docs/UserGuide.md:213]: is this true?

**Review by purivirakarin — APPROVED** [review]

**Review by shaunlxw — APPROVED** [review]
LGTM! 👍

**Pull request merged**